libcontainer/container_linux.go

```diff
@@ -775,6 +775,27 @@ func (c *linuxContainer) updateState(process parentProcess) error {
 	return json.NewEncoder(f).Encode(state)
 }
 
+func (c *linuxContainer) checkFreezer() (Status, error) {
+	path := c.cgroupManager.GetPaths()["freezer"]
+	if _, err := os.Stat(path); err != nil {
+		if os.IsNotExist(err) {
+			// Freezer subsystem is not available,Application is neither Checkpointed nor Destroyed
+			// return the status as Running
+			return Running, nil
+		}
+	}
+	contents, err := ioutil.ReadFile(filepath.Join(path, "freezer.state"))
+	if err != nil {
+		return 0, newSystemError(err)
+	}
+	freezerstate := string(contents)
+	if strings.TrimSpace(freezerstate) == "FROZEN" {
```

Comment:
Maybe we should check both `FROZEN` and `FREEZING` here, we have `pausing` defined but never used, it should be returned here.